server/handler/base.go

```diff
@@ -268,6 +271,29 @@ func (b *Base) EvaluateFetchedConfig(ctx context.Context, prctx pull.Context, cl
 	return result, nil
 }
 
+func (b *Base) ValidateChangedHeadConfig(ctx context.Context, prctx pull.Context, client *github.Client, trigger common.Trigger) error {
+	logger := zerolog.Ctx(ctx)
+
+	files, err := prctx.ChangedFiles()
+	if err != nil {
+		return err
+	}
+	for _, f := range files {
+		switch f.Status {
+		case pull.FileAdded, pull.FileModified:
+			switch f.Filename {
+			case b.PullOpts.PolicyPath, ".github/" + b.PullOpts.PolicyPath:
+				logger.Debug().Msg("Pull request changed policy file. Evaluating head version.")
+
+				fetchedConfig := b.ConfigFetcher.ConfigForPRHead(ctx, prctx, client)
+				_, err = b.ValidateFetchedConfig(ctx, prctx, client, fetchedConfig, trigger)
+				return err
```

Comment:
I'm not sure if the error returned from `ValidateFetchedConfig` makes it clear where the error was. We may want to add an additional message to clarify that this is from the policy being modified in the pull request, rather than the existing policy in the repository.